webapp/nuxt.config.js

```diff
@@ -277,6 +277,26 @@ export default {
           },
         ],
       })
+      const tagAttributesForTesting = ['data-test', ':data-test', 'v-bind:data-test']
+      ctx.loaders.vue.compilerOptions = {
+        modules: [
+          {
+            preTransformNode(abstractSyntaxTreeElement) {
+              if (!ctx.isDev) {
+                const { attrsMap, attrsList } = abstractSyntaxTreeElement
+                tagAttributesForTesting.forEach(attribute => {
+                  if (attrsMap[attribute]) {
+                    delete attrsMap[attribute]
+                    const index = attrsList.findIndex(attr => attr.name === attribute)
+                    attrsList.splice(index, 1)
+                  }
+                })
+              }
+              return abstractSyntaxTreeElement
```

Comment:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-12-10T12:21:43Z" title="Tuesday, December 10th 2019, 1:21:43 pm +01:00">Dec 10, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -272,6 +272,26 @@ export default {
           },
         },
       })
+      const tagAttributesForTesting = ['data-test', ':data-test', 'v-bind:data-test']
+      ctx.loaders.vue.compilerOptions = {
+        modules: [
+          {
+            preTransformNode(astEl) {
+              if (!ctx.isDev) {
+                const { attrsMap, attrsList } = astEl
+                tagAttributesForTesting.forEach(attribute => {
+                  if (attrsMap[attribute]) {
+                    delete attrsMap[attribute]
+                    const index = attrsList.findIndex(attr => attr.name === attribute)
+                    attrsList.splice(index, 1)
+                  }
+                })
+              }
+              return astEl
```


What does `astEl` stand for? Could we use a more declarative name?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-10T12:39:27Z" title="Tuesday, December 10th 2019, 1:39:27 pm +01:00">Dec 10, 2019</time>_

---


I'm not sure... this was a copy :spaghetti: and once it was working, I thought about how it could be improved, but didn't spend much time cause I need to build the image each time to make sure it works.... 
I learned about `delete` from @roschaefer actually. We have some cases in the backend already, really cool stuff!!
this is what `astEl` looks like 
```
{                                                                                                                                                               13:35:28
  type: 1,
  tag: 'signup',
  attrsList: [
    {
      name: ':invitation',
      value: 'true'
    }
  ],
  attrsMap: {
    ':invitation': 'true'
  },
  rawAttrsMap: {},
  parent: {
    type: 1,
    tag: 'ds-card',
    attrsList: [
      [Object]
    ],
    attrsMap: {
      class: 'signup'
    },
    rawAttrsMap: {},
    parent: {
      type: 1,
      tag: 'ds-section',
      attrsList: [],
      attrsMap: {},
      rawAttrsMap: {},
      parent: undefined,
      children: [Array]
    },
    children: []
  },
  children: []
}
```
Any ideas of names? `El` I guess would be short for `Element`... not sure about `ast`...

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-10T12:53:45Z" title="Tuesday, December 10th 2019, 1:53:45 pm +01:00">Dec 10, 2019</time>_

---


Awww, I was able to find out... it stands for `abstractSyntaxTree`... have updated the code to be more verbose... I love it when people ask for changes I could see myself suggesting!! :heart_eyes_cat:

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-12-10T14:09:01Z" title="Tuesday, December 10th 2019, 3:09:01 pm +01:00">Dec 10, 2019</time>_

---


Thank you for researching! 😍